modules/standard/IO.chpl

```diff
@@ -4022,6 +4040,83 @@ proc channel.writebits(v:integral, nbits:integral):bool throws {
   return try this.write(new ioBits(v:uint(64), nbits:int(8)));
 }
 
+proc channel.writeBinary(arg:numeric, param endian:ioendian = ioendian.native) throws {
+  var kind: iokind;
+  var e:syserr = ENOERR;
+
+  select (endian) {
+    when ioendian.native {
+      e = try _write_binary_internal(_channel_internal, iokind.native, arg);      
+    }
+    when ioendian.big {
+      e = try _write_binary_internal(_channel_internal, iokind.big, arg);
+    }
+    when ioendian.little {
+      e = try _write_binary_internal(_channel_internal, iokind.little, arg);
+    }
+  }
+  if (e != ENOERR) {
+    throw SystemError.fromSyserr(e);
+  }
+}
+
+proc channel.writeBinary(arg:numeric, endian:ioendian) throws {
+  var kind: iokind;
+
+  select (endian) {
+    when ioendian.native {
+      this.writeBinary(arg, ioendian.native);
+    }
+    when ioendian.big {
+      this.writeBinary(arg, ioendian.big);
+    }
+    when ioendian.little {
+      this.writeBinary(arg, ioendian.little);
+    }
+  }
+}
+
+proc channel.readBinary(ref arg:numeric, param endian:ioendian = ioendian.native):bool throws {
+  var rv: bool = false;
+  var e:syserr = ENOERR;
+
+  select (endian) {
+    when ioendian.native {
+      e = try _read_binary_internal(_channel_internal, iokind.native, arg);      
```

Comment:
Ditto here about how to potentially avoid the select statement to call `_read_binary_internal`